Userland/Libraries/LibWeb/HTML/Parser/StackOfOpenElements.cpp

```diff
@@ -68,9 +68,22 @@ bool StackOfOpenElements::has_in_list_item_scope(const FlyString& tag_name) cons
     return has_in_scope_impl(tag_name, list);
 }
 
+// The stack of open elements is said to have a particular element in select scope
+// when it has that element in the specific scope consisting of all element types except the following:
+// - optgroup in the HTML namespace
+// - option in the HTML namespace
+// NOTE: In this case it's "all element types _except_"
 bool StackOfOpenElements::has_in_select_scope(const FlyString& tag_name) const
 {
-    return has_in_scope_impl(tag_name, { "option", "optgroup" });
+    for (ssize_t i = m_elements.size() - 1; i >= 0; --i) {
+        auto& node = m_elements.at(i);
+        if (node.local_name() == tag_name)
+            return true;
+        if (node.local_name() == HTML::TagNames::option || node.local_name() == HTML::TagNames::optgroup)
+            continue;
+        return false;
+    }
+    VERIFY_NOT_REACHED();
 }
```

Comment:
Is there a Spec link for this?
(Also I would place a comma after the first line)